sql/executor.go

```diff
@@ -39,11 +42,18 @@ var errTransactionInProgress = errors.New("there is already a transaction in pro
 type Executor struct {
 	db     client.DB
 	nodeID uint32
+
+	// System Config and mutex.
+	systemConfig   *config.SystemConfig
```

Comment:
Perhaps use `unsafe.Pointer` and `atomic.StorePointer` and `atomic.LoadPointer`. I don't expect the performance to be better, but it might be clearer what is going on and would prevent accidental access of `systemConfig` without holding `systemConfigMu`.